pkg/controller/nodeipam/ipam/multinetwork_cloud_cidr_allocator.go

```diff
@@ -82,3 +81,44 @@ func resourceName(name string) string {
 	parts := strings.Split(name, "/")
 	return parts[len(parts)-1]
 }
+
+func (ca *cloudCIDRAllocator) NetworkToNodes(network *networkv1.Network) error {
+	k8sNodesList, err := ca.nodeLister.List(labels.Everything())
+	if err != nil {
+		return fmt.Errorf("error fetching nodes: %v", err)
+	}
+	// Need to reschedule all Nodes, since we do not know which one holds the
+	// new Network.
+	for _, node := range k8sNodesList {
+		if network != nil {
+			// filter out nodes that are not part of network
+			if node.Annotations == nil {
+				// skip node w/o any annotation, not possible for it to have any MN
+				continue
+			}
+			northIntfAnn, ok := node.Annotations[networkv1.NorthInterfacesAnnotationKey]
+			if !ok {
+				// skip node w/o "north-interfaces" annotation, no MN
+				continue
+			}
+			northIntf, err := networkv1.ParseNorthInterfacesAnnotation(northIntfAnn)
+			// if err!=nil means there is some format issue with the annotation, lets
```

Comment:
that doesn't seem correct, the controller logic reconcile and if is not able to do it it requeue the same object to reconcile, so the loop is idempotent and guarantee this.

If err is != nil and the annotation is wrong, something is very wrong down the stack and we should not modify it, because there is a high risk of hotlooping, since the controller that added it will need to move it to the status it added

Reply:
This is not the reconcile loop. This function translates the trigger from Network to Node. So if this function fails, there will not be a retry.

Reply:
oh, you are processing directly on the handlers 🤒